### PR TITLE
Docker doc

### PR DESCRIPTION
**Pull request by enderm**

### CODE REVIEW

File: docs/user/DockerUsage.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+- Make sure you have Docker [installed on your workstation](../../README.md#docker).
+
+- Prepare a file with authentication info, as described in [Authenticating Terraform to access Azure](./TerraformAzureAuthentication.md)
+
+## Build the docker image
+
+Run the following command to create the `viya4-iac-azure` docker image that will be used in subsequent steps:
+
+```bash
+docker build -t viya4-iac-azurwe .
```
Comment (thpang) [inline docs/user/DockerUsage.md:14]: Typo : viya4-iac-azurwe -> viya4-iac-azure

File: main.tf [hunk 2]
```diff
@@ -395,33 +412,22 @@ data "external" "iac_tooling_version" {
...
-  template = file("${path.module}/files/sas-iac-buildinfo.yaml.tmpl")
-  vars = {
-    git-hash            = lookup(data.external.git_hash.result, "git-hash")
-    timestamp           = chomp(timestamp())
-    iac-tooling         = var.iac_tooling
-    terraform-version   = lookup(data.external.iac_tooling_version.result, "terraform_version")
-    provider-selections = lookup(data.external.iac_tooling_version.result, "provider_selections")
-    terraform-revision  = lookup(data.external.iac_tooling_version.result, "terraform_revision")
-    terraform-outdated  = lookup(data.external.iac_tooling_version.result, "terraform_outdated")
+resource "kubernetes_config_map" "sas_iac_buildinfo" {
```
Comment (thpang) [inline main.tf:415]: Again, not sure why we're changing patterns here? We use a template file and now a data path? Would like a discussion on this change.

**Comment by thpang** [comment]
Also, more changes here than just doc updates.

**Comment by enderm** [comment]
the code changes were necessary to allow the docker examples to work as documented (primarily to enable to "-u" syntax)

File: main.tf (hunk 2 above)
Comment (enderm) [reply main.tf:415]: change needed to avoid permission issue with local file when running docker with local user.
Plus, it's a code simplification, imo

Comment (thpang) [reply main.tf:415]: Ok, we need to have a meeting to talk about this. Thanks.

**Pull request merged**